src/jit-analyze/Program.cs

```diff
@@ -55,140 +40,17 @@ public class jitanalyze
-            }
+            _command = command;
 
-            private void Validate()
-            {
-                if (_basePath == null)
-                {
-                    _syntaxResult.ReportError("Base path (--base) is required.");
-                }
-
-                if (_diffPath == null)
-                {
-                    _syntaxResult.ReportError("Diff path (--diff) is required.");
-                }
-
-                if (_metrics == null)
-                {
-                    _metrics = new List<string> { "CodeSize" };
-                }
-
-                foreach (string metricName in _metrics)
-                {
-                    if (!MetricCollection.ValidateMetric(metricName))
-                    {
-                        _syntaxResult.ReportError($"Unknown metric '{metricName}'. Available metrics: {MetricCollection.ListMetrics()}");
-                    }
-                }
-
-                if (OverrideTotalBaseMetric.HasValue != OverrideTotalDiffMetric.HasValue)
-                {
-                    _syntaxResult.ReportError("override-total-base-metric and override-total-diff-metric must either both be specified or both not be specified");
-                }
-            }
-
-            public string BasePath { get { return _basePath; } }
-            public string DiffPath { get { return _diffPath; } }
-            public bool Recursive { get { return _recursive; } }
-            public string FileExtension { get { return _fileExtension; } }
-            public bool Full { get { return _full; } }
-            public bool Warn { get { return _warn; } }
-            public int Count { get { return _count; } }
-            public string TSVFileName { get { return _tsv; } }
-            public string JsonFileName { get { return _json; } }
-            public string MarkdownFileName { get { return _md;  } }
-            public bool DoGenerateJson { get { return _json != null; } }
-            public bool DoGenerateTSV { get { return _tsv != null; } }
-            public bool DoGenerateMarkdown { get { return _md != null; } }
-            public bool Reconcile { get { return !_noreconcile; } }
-            public string Note { get { return _note; } }
-            public double? OverrideTotalBaseMetric => _overrideTotalBaseMetric;
-            public double? OverrideTotalDiffMetric => _overrideTotalDiffMetric;
-            public bool IsDiffsOnly => _isDiffsOnly;
-            public bool IsSubsetOfDiffs => _isSubsetOfDiffs;
-
-            public string Filter {  get { return _filter; } }
-
-            public List<string> Metrics {  get { return _metrics; } }
-            public bool SkipTextDiff { get { return _skipTextDiff;  } }
-            public bool RetainOnlyTopFiles { get { return _retainOnlyTopFiles; } }
+            _reconcile = !Get(command.NoReconcile);
+            _filter = Get(command.Filter);
+            _overrideTotalBaseMetric = Get(command.OverrideTotalBaseMetric);
```

Comment:
Does this properly parse with invariant culture?

Reply:
Good catch! I have added a custom invariant culture parser for `Option<double>`, with the error message which they normally show (e.g. for `int` options).